pandas/core/style.py

```diff
@@ -144,7 +151,18 @@ def __init__(self, data, precision=None, table_styles=None, uuid=None,
         self.table_styles = table_styles
         self.caption = caption
         if precision is None:
-            precision = pd.options.display.precision
+            precision = {'__default__': pd.options.display.precision}
```

Comment:
just call ``.set_precision`` instead of repeating this (and all of this logic should be there).

Further I think that a ``.set_precision()`` should reset to the default (e.g. ``pd.options.display.precision``), so pls put in a test for this.